### PR TITLE
feature/filter-orders

### PR DESCRIPTION
**Pull request by rniedziolka**
Hi,
the next feature is ready:
https://www.loom.com/share/4451ede18ed343dfa6b4809df950a11d
Waiting for the feedback!

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -2,15 +2,19 @@
 
 class OrdersController < ApplicationController
   def index
-    @orders = Order.where(employee: current_employee)
+    if params[:status].present?
```
Comment (ciejj) [inline app/controllers/orders_controller.rb]: what happens, if params[:status]='foo'?

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -2,15 +2,19 @@
...
+      render :index, locals: { orders: Order.where(employee: current_employee).filter_by_status(params[:status]) }
+    else
+      render :index, locals: { orders: Order.where(employee: current_employee) }
```
Comment (ciejj) [inline app/controllers/orders_controller.rb:8]: I would consider moving rendering outside if-else and passing default filtering instead

File: app/views/pages/_headeradmin.html.erb [hunk 3]
```diff
@@ -23,7 +23,7 @@
 
     <span type="button">
     <a href="/admin/orders"
-     class="btn btn-outline-success">Orders (<%= Order.count - Order.count(&:delivered) %>) </a>
+     class="btn btn-outline-success">Orders (<%= Order.where("status = '0'").count %>) </a>
```
Comment (ciejj) [inline app/views/pages/_headeradmin.html.erb]: Try using enum helper method here

File: spec/system/order_filter_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+require 'rails_helper'
+
+RSpec.describe 'Order filtering', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  let!(:employee) { create(:employee) }
+  let!(:order1) { create(:order, employee: employee, status: 'delivered') }
```
Comment (ciejj) [inline spec/system/order_filter_spec.rb]: Consider naming those orders: :order_delivered, :order_not_delivered - it may make specs easier to read

**Review by ciejj — COMMENTED** [review]
Nice job, but we can improve it slightly :)

File: app/controllers/orders_controller.rb (hunk 1 above)
Comment (rniedziolka) [reply app/controllers/orders_controller.rb]: will, be an error - ups, sorry for that
can I use this line?
if ["delivered", "not_delivered"].include?params[:status]

Comment (rniedziolka) [reply app/controllers/orders_controller.rb]: Rubocop corrected me :D
if %w[delivered not_delivered].include?(params[:status])
so fixed :)

File: app/views/pages/_headeradmin.html.erb (hunk 3 above)
Comment (rniedziolka) [reply app/views/pages/_headeradmin.html.erb]: Yes, you are right, fixed.

File: spec/system/order_filter_spec.rb (hunk 4 above)
Comment (rniedziolka) [reply spec/system/order_filter_spec.rb]: True, fixed.

File: app/controllers/orders_controller.rb (hunk 2 above)
Comment (rniedziolka) [reply app/controllers/orders_controller.rb:8]: I've tried to do it, but I don't know how to syntactically work it out. @ciejj Could you give me a hint?

Comment (ciejj) [reply app/controllers/orders_controller.rb:8]: let's leave it like it is right now - we will refactor it later, when more sorting options will be available - then the reason for refactor will be easier to understand

**Review by ciejj — APPROVED** [review]
👍

**Review by oskarlakner — APPROVED** [review]

**Pull request merged**